cmd/relay/bgs/bgs.go

```diff
@@ -0,0 +1,1375 @@
+
+	admin := e.Group("/admin", bgs.checkAdminAuth)
+
+	// Slurper-related Admin API
+	admin.GET("/subs/getUpstreamConns", bgs.handleAdminGetUpstreamConns)
+	admin.GET("/subs/getEnabled", bgs.handleAdminGetSubsEnabled)
+	admin.GET("/subs/perDayLimit", bgs.handleAdminGetNewPDSPerDayRateLimit)
+	admin.POST("/subs/setEnabled", bgs.handleAdminSetSubsEnabled)
+	admin.POST("/subs/killUpstream", bgs.handleAdminKillUpstreamConn)
+	admin.POST("/subs/setPerDayLimit", bgs.handleAdminSetNewPDSPerDayRateLimit)
+
+	// Domain-related Admin API
+	admin.GET("/subs/listDomainBans", bgs.handleAdminListDomainBans)
+	admin.POST("/subs/banDomain", bgs.handleAdminBanDomain)
+	admin.POST("/subs/unbanDomain", bgs.handleAdminUnbanDomain)
+
+	// Repo-related Admin API
+	admin.POST("/repo/takeDown", bgs.handleAdminTakeDownRepo)
+	admin.POST("/repo/reverseTakedown", bgs.handleAdminReverseTakedown)
+	admin.GET("/repo/takedowns", bgs.handleAdminListRepoTakeDowns)
+
+	// PDS-related Admin API
+	admin.POST("/pds/requestCrawl", bgs.handleAdminRequestCrawl)
+	admin.GET("/pds/list", bgs.handleListPDSs)
+	admin.POST("/pds/changeLimits", bgs.handleAdminChangePDSRateLimits)
+	admin.POST("/pds/block", bgs.handleBlockPDS)
+	admin.POST("/pds/unblock", bgs.handleUnblockPDS)
+	admin.POST("/pds/addTrustedDomain", bgs.handleAdminAddTrustedDomain)
+
+	// Consumer-related Admin API
+	admin.GET("/consumers/list", bgs.handleAdminListConsumers)
+
+	// In order to support booting on random ports in tests, we need to tell the
+	// Echo instance it's already got a port, and then use its StartServer
+	// method to re-use that listener.
+	e.Listener = listen
+	srv := &http.Server{}
+	return e.StartServer(srv)
+}
+
+func (bgs *BGS) Shutdown() []error {
+	errs := bgs.slurper.Shutdown()
+
+	if err := bgs.events.Shutdown(context.TODO()); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errs
+}
+
+type HealthStatus struct {
+	Status  string `json:"status"`
+	Message string `json:"msg,omitempty"`
+}
+
+func (bgs *BGS) HandleHealthCheck(c echo.Context) error {
+	if err := bgs.db.Exec("SELECT 1").Error; err != nil {
+		bgs.log.Error("healthcheck can't connect to database", "err", err)
+		return c.JSON(500, HealthStatus{Status: "error", Message: "can't connect to database"})
+	} else {
+		return c.JSON(200, HealthStatus{Status: "ok"})
+	}
+}
+
+var homeMessage string = `
+d8888b. d888888b  d888b  .d8888. db   dD db    db
+88  '8D   '88'   88' Y8b 88'  YP 88 ,8P' '8b  d8'
+88oooY'    88    88      '8bo.   88,8P    '8bd8'
+88~~~b.    88    88  ooo   'Y8b. 88'8b      88
+88   8D   .88.   88. ~8~ db   8D 88 '88.    88
+Y8888P' Y888888P  Y888P  '8888Y' YP   YD    YP
+
+This is an atproto [https://atproto.com] relay instance, running the 'bigsky' codebase [https://github.com/bluesky-social/indigo]
+
+The firehose WebSocket path is at:  /xrpc/com.atproto.sync.subscribeRepos
+`
+
+func (bgs *BGS) HandleHomeMessage(c echo.Context) error {
+	return c.String(http.StatusOK, homeMessage)
+}
+
+type AuthToken struct {
```

Comment:
Storing auth tokens in the database like this is old code.

Do we need this functionality? Seems like it could just be an env var, maybe with multiple values.

Reply:
yeah, there's an env var. currently it sets the db; we could just hold it in memory and stop bouncing off the db; I'd be pretty happy about that actually....